static/js/apply-for-jobs.js

```diff
@@ -107,47 +111,51 @@
     educationContainer.appendChild(educationInput);
   };
 
-  schoolsList.addEventListener("input", debounce(async (e) => {
-    const datalistElement = document.querySelector("#school-0");
-    var schools = [];
-    if (e.target.value !== "") {
-      schools = await getDataFromGreenhouseApi("schools", e.target.value);
-    }
-    var list = "";
-    schools.forEach(school => {
-      list += `<option value="${school.text}" />`
-    });
-    datalistElement.innerHTML = list;
-  }, 350));
-
-  addEducationButton.addEventListener("click", () => {
-    numberOfEducations++;
-    addEducationInput(numberOfEducations);
-    // Add click event listener to the "remove education" button
-    const removeEducationButtons = document.querySelectorAll(".js-remove-education");
-    removeEducationButtons[removeEducationButtons.length - 1].addEventListener("click", (e) => {
-      e.preventDefault();
-      const educationToBeRemoved = document.querySelector(`[data-education='${e.target.parentElement.parentElement.dataset.education}']`);
-      educationToBeRemoved.parentNode.removeChild(educationToBeRemoved);
-    });
-    // Populate new added input datalists
-    populateDegreeList(`#degree-${numberOfEducations}`, defaultDegreeList);
-    populateDisciplineList(`#discipline-${numberOfEducations}`, defaultDisciplineList);
-    // Add input event listener to the new school select
-    const newSchoolsList = document.querySelector(`.js-school-${numberOfEducations}`);
-    newSchoolsList.addEventListener("input", debounce(async (e) => {
-      const datalistElement = document.querySelector(`#school-${numberOfEducations}`);
+  if (schoolsList) {
+    schoolsList.addEventListener("input", debounce(async (e) => {
+      const datalistElement = document.querySelector("#school-0");
       var schools = [];
-      if (newSchoolsList.value !== "") {
-        schools = await getDataFromGreenhouseApi("schools", newSchoolsList.value);
+      if (e.target.value !== "") {
+        schools = await getDataFromGreenhouseApi("schools", e.target.value);
       }
       var list = "";
       schools.forEach(school => {
         list += `<option value="${school.text}" />`
       });
-      datalistElement.innerHTML = list
+      datalistElement.innerHTML = list;
     }, 350));
-  });
+  }
+
+  if (addEducationButton) {
+    addEducationButton.addEventListener("click", () => {
+      numberOfEducations++;
+      addEducationInput(numberOfEducations);
+      // Add click event listener to the "remove education" button
+      const removeEducationButtons = document.querySelectorAll(".js-remove-education");
+      removeEducationButtons[removeEducationButtons.length - 1].addEventListener("click", (e) => {
+        e.preventDefault();
+        const educationToBeRemoved = document.querySelector(`[data-education='${e.target.parentElement.parentElement.dataset.education}']`);
```

Comment:
This is very reliant on the DOM staying the same. Can you use `closest` to negate that? https://developer.mozilla.org/en-US/docs/Web/API/Element/closest